### PR TITLE
Fix issues/pulls dependencies problems

### LINKED ISSUE #6039: Pullrequest dependencies are disabled when internal issue tracker is disabled
- Gitea version (or commit ref): master

## Description

When selecting an external issue tracker in the repository settings (see 1st screenshot), it is no longer possible to add dependencies between pull requests.

In our current workflow, we do use an external issue tracker for all of our repositories.

However, we're leveraging PRs and would love to be able to have dependencies between PRs (which does work when internal issue tracker is enabled).

## Screenshots

grafik

### PR DESCRIPTION
**Pull request by lunny**
Fix #6039

### CODE REVIEW

**Comment by lunny** [comment]
I also found more problems about issue dependencies and will fix them on next PRs.

**Review by 6543 — APPROVED** [review]
Need `make generate-swagger` beside that ok

File: routers/api/v1/repo/issue.go [hunk 1]
```diff
@@ -51,6 +51,11 @@ func SearchIssues(ctx *context.APIContext) {
 	//   description: repository to prioritize in the results
 	//   type: integer
 	//   format: int64
+	// - name: tp
```
Comment (silverwind) [inline routers/api/v1/repo/issue.go]: Is "tp" supposed to mean "type"? If so, suggesting rename for clarity.

File: routers/api/v1/repo/issue.go [hunk 2]
```diff
@@ -51,6 +51,11 @@ func SearchIssues(ctx *context.APIContext) {
...
+	//   in: query
+	//   description: both issues and pull requests or one of them, could be prs, issues or blank
+	//   type: integer
```
Comment (silverwind) [inline routers/api/v1/repo/issue.go]: You are checking for string below, I think this is wrong.

File: routers/api/v1/repo/issue.go (hunk 1 above)
Comment (6543) [reply routers/api/v1/repo/issue.go]: hmm you are right - for non gita familiar people it looks weared ..

**Review by 6543 — CHANGES_REQUESTED** [review]
make generate-swagger + rename filed + clarify description

File: routers/api/v1/repo/issue.go [hunk 3]
```diff
@@ -51,6 +51,11 @@ func SearchIssues(ctx *context.APIContext) {
 	//   description: repository to prioritize in the results
 	//   type: integer
 	//   format: int64
+	// - name: tp
+	//   in: query
+	//   description: both issues and pull requests or one of them, could be prs, issues or blank
```
Comment (6543) [inline routers/api/v1/repo/issue.go]:
```suggestion
	//   description: filter by type (issues / pull requests) if set
```

**Comment by lunny** [comment]
@silverwind @6543 done.

**Review by 6543 — APPROVED** [review]

File: web_src/js/index.js [hunk 4]
```diff
@@ -3455,9 +3455,10 @@ function initIssueList() {
   const repolink = $('#repolink').val();
   const repoId = $('#repoId').val();
   const crossRepoSearch = $('#crossRepoSearch').val();
-  let issueSearchUrl = `${suburl}/api/v1/repos/${repolink}/issues?q={query}`;
+  const tp = $('#tp').val();
+  let issueSearchUrl = `${suburl}/api/v1/repos/${repolink}/issues?q={query}&tp=${tp}`;
   if (crossRepoSearch === 'true') {
-    issueSearchUrl = `${suburl}/api/v1/repos/issues/search?q={query}&priority_repo_id=${repoId}`;
+    issueSearchUrl = `${suburl}/api/v1/repos/issues/search?q={query}&priority_repo_id=${repoId}&tp=${tp}`;
```
Comment (silverwind) [inline web_src/js/index.js]: Need to update this file to `type`.

File: routers/api/v1/repo/issue.go [hunk 5]
```diff
@@ -128,6 +133,16 @@ func SearchIssues(ctx *context.APIContext) {
 		}
 	}
 
+	var isPull util.OptionalBool
+	switch ctx.Query("tp") {
+	case "pulls":
```
[the remark this answers is not in the document]
Comment (silverwind) [reply routers/api/v1/repo/issue.go:137]: Yeah, also "issue". Of course, ignore if plural is already in use somewhere else.

[the remark this answers is not in the document]
Comment (lunny) [reply routers/api/v1/repo/issue.go:137]: I think most place are using `issues` and `pulls` as `issueType`.

File: web_src/js/index.js (hunk 4 above)
Comment (lunny) [reply web_src/js/index.js]: done.

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 6]
```diff
@@ -428,6 +428,7 @@
 	<input type="hidden" id="repolink" value="{{$.RepoRelPath}}">
 	<input type="hidden" id="repoId" value="{{.Repository.ID}}">
 	<input type="hidden" id="crossRepoSearch" value="{{.AllowCrossRepositoryDependencies}}">
+	<input type="hidden" id="type" value="{{.IssueType}}">
 	<!-- I know, there is probably a better way to do this -->
```
Comment (guillep2k) [inline templates/repo/issue/view_content/sidebar.tmpl:432]: Not relevant for this PR, but we should get rid of this comment or at least hide it from the rendered output:

```suggestion
	{{/* I know, there is probably a better way to do this */}}
```

**Review by guillep2k — APPROVED** [review]

**Pull request merged**